src/commands/run.rs

```diff
@@ -19,18 +25,64 @@ impl Run {
         port: u32,
         filter_search_domain: String,
     ) -> Result<(), Error> {
-        debug!(
-            "Setting up aardvark server with input directory as {:?}",
-            input_dir
-        );
+        // create a temporary path for unix socket
+        // so parent can communicate with child and
+        // only exit when child is ready to serve.
+        let (ready_pipe_read, ready_pipe_write) = nix::unistd::pipe()?;
 
-        if let Err(er) = serve::serve(&input_dir, port, &filter_search_domain) {
-            return Err(std::io::Error::new(
-                std::io::ErrorKind::Other,
-                format!("Error starting server {}", er),
-            ));
+        // fork and verify if server is running
+        // and exit parent
+        // setsid() ensures that there is no controlling terminal on the child process
+        match unsafe { fork() } {
```

Comment:
Why is this unsafe?

Reply:
`nix`'s fork is returns unsafe most likely because it invokes `fork(` directly in such cases rust compiler will bark however our use-case is okay here so all such functions can be invoked under `unsafe`. See https://docs.rs/nix/latest/nix/unistd/fn.fork.html and this https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html

But invocation is fine for our use-case.